### PR TITLE
apply dependency management

### PR DESCRIPTION
**Pull request by qiaouchicago**
https://jira.opensciencedatacloud.org/browse/DEV-280

### CODE REVIEW

**Comment by qiaouchicago** [comment]
Used separate requirements.txt for py2 and py3, because py2 is using functools32.

**Comment by qiaouchicago** [comment]
.travis.yaml not removed because of no internal dependencies.

File: tox.ini [hunk 1]
```diff
@@ -3,7 +3,7 @@ envlist = py27, py36
 
 [testenv]
 deps =
-    -rrequirements.txt
+    setuptools==44.1.1
```
Comment (kulgan) [inline tox.ini:6]: This applies to py27 alone right? If yes, maybe you can place it under `[testenv:py2]` instead

File: tox.ini [hunk 2]
```diff
@@ -3,7 +3,13 @@ envlist = py27, py36
...
     -rdev-requirements.txt
 commands = 
     pytest -lvv --cov=gdcdictionary --cov-report xml --cov-report html --junit-xml test-reports/results.xml  {posargs}
+
+[testenv:py27]
+deps =
+    setuptools==44.1.1
+    -rdev-requirements.txt
+commands = 
+    pytest -lvv --cov=gdcdictionary --cov-report xml --cov-report html --junit-xml test-reports/results.xml  {posargs}
```
Comment (jiakf) [inline tox.ini:15]: EOF new line?

**Review by jiakf — APPROVED** [review]
lgtm

File: setup.py [hunk 3]
```diff
@@ -9,8 +9,9 @@
     },
     setup_requires=['setuptools_scm'],
     install_requires=[
-        'PyYAML',
-        'jsonschema',
+        'PyYAML>=5.1.2,<5.5',
+        'jsonschema>=3.0.2,<3.3',
+        'pyrsistent~=0.16.1'        # pin to version 0.16 for py27
```
Comment (kulgan) [inline setup.py:14]: Will [environmental markers](https://www.python.org/dev/peps/pep-0508/#environment-markers) work for this? eg 
```suggestion
        'pyrsistent~=0.16.1';python_version="2.7"
```

**Review by kulgan — APPROVED** [review]

**Review by ProfOak — APPROVED** [review]

**Pull request merged**